### PR TITLE
BUGFIX: RAIL-4720 Save button behaviour

### PR DESCRIPTION
**Pull request by kandl**
In some cases, when you click the save button
(for example immediately after you insert an insight) and the dashboard is still processing some requests related to this insight, it can take few seconds,
before the CMD.SAVE starts to process
(as dashboard can process only 1 command at a time). We want to show the loading indicator immediately and avoid any double-save race conditions, so add a local state for this case.

JIRA: RAIL-4720

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/button/saveButton/DefaultSaveButton.tsx [hunk 1]
```diff
@@ -32,25 +32,33 @@ export function useSaveButtonProps(): ISaveButtonProps {
...
+
+    const onSaveClick = useCallback(() => {
+        setOptimisticIsSaving(true);
+        dispatchAndWaitFor(
+            dispatch,
+            // the || is intentional, we want to replace empty string as well
+            saveDashboard(title || emptyTitle),
+        ).then(() => {
+            setOptimisticIsSaving(false);
+            dispatch(cancelEditRenderMode());
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/button/saveButton/DefaultSaveButton.tsx:50]: What about `catch`? We should cancel the optimistic saving there as well, right?

Comment (kandl) [reply libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/button/saveButton/DefaultSaveButton.tsx:50]: I add it, but I think when the save will fail, it wont work anyway :)

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10910/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12997/ : SUCCESS in 12m 09s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3346

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/212/ : FAILURE in 1m 53s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3478/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2327/ : SUCCESS in 17m 25s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12998/ : SUCCESS in 14m 01s

**Pull request merged**